### ISSUE TITLE
Crash in channel.rb:260

### ISSUE THREAD

**Issue by tru** [zerowidth/camper_van#11]
```

2012-04-25 14:30:02  5670 DEBUG CamperVan : irc <- :camper_van PONG camper_van
2012-04-25 14:30:02  5670 DEBUG CamperVan : irc -> MODE #plex_incorporated b
2012-04-25 14:30:02  5670 DEBUG CamperVan : irc <- :camper_van 472 tobias_hieta b :Unknown mode b
2012-04-25 14:30:34  5670  WARN CamperVan : WARNING: Tried to set starred= "false" on a Firering::Message instance but it didn't respond. Probably the API got updated, please report this! (https://github.com/EmmanuelOga/firering/issues)
/var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:388: warning: multiple values for a block parameter (2 for 1)
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/data/message.rb:46
2012-04-25 14:30:34  5670  INFO CamperVan : closed connection from 127.0.0.1
2012-04-25 14:30:34  5670 ERROR CamperVan : http error . Trying again in 2 seconds...
2012-04-25 14:30:34  5670 ERROR CamperVan : http error . Trying again in 2 seconds...
2012-04-25 14:30:34  5670 ERROR CamperVan : http error . Trying again in 2 seconds...
/var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:260:in `map_message_to_irc': undefined method `name' for #<Array:0x7fdda51ce780> (NoMethodError)
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:389:in `user_for_message'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/data/message.rb:46:in `call'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/data/message.rb:46:in `user'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:388:in `user_for_message'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:257:in `map_message_to_irc'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:247:in `stream_campfire_to_channel'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/connection.rb:84:in `call'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/connection.rb:84:in `stream'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/connection.rb:97:in `call'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/connection.rb:97:in `<<'
    from /var/lib/gems/1.8/gems/firering-1.1.1/lib/firering/connection.rb:97:in `stream'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:275:in `call'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:275:in `on_decoded_body_data'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:268:in `on_body_data'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:582:in `process_chunk_body'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:330:in `dispatch'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:256:in `receive_data'
    from /var/lib/gems/1.8/gems/eventmachine-0.12.10/lib/eventmachine.rb:256:in `run_machine'
    from /var/lib/gems/1.8/gems/eventmachine-0.12.10/lib/eventmachine.rb:256:in `run'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/server.rb:21:in `run'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/bin/camper_van:64
    from /usr/local/bin/camper_van:19:in `load'
    from /usr/local/bin/camper_van:19
```

**Comment by zerowidth** [comment]
Please update your firering gem to the latest version and try again -- looks like I overlooked the dependency update last release, so you weren't forced to upgrade with the latest camper_van version.

**Comment by tru** [comment]
I updated it (gem install firering; git cloned; gem build *gemspec; gem install *gem) and I get something similar still:

```
2012-04-26 09:28:09 26277  INFO CamperVan : performing request to https://osxbmc.campfirenow.com/room/275936/speak.json
/var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:388: warning: multiple values for a block parameter (2 for 1)
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/data/message.rb:46
2012-04-26 09:28:11 26277  INFO CamperVan : closed connection from 127.0.0.1
2012-04-26 09:28:11 26277 ERROR CamperVan : http error . Trying again in 2 seconds...
2012-04-26 09:28:11 26277 ERROR CamperVan : http error . Trying again in 2 seconds...
2012-04-26 09:28:11 26277 ERROR CamperVan : http error . Trying again in 2 seconds...
/var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:260:in `map_message_to_irc': undefined method `name' for #<Array:0x7f8cded4d178> (NoMethodError)
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:389:in `user_for_message'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/data/message.rb:46:in `call'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/data/message.rb:46:in `user'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:388:in `user_for_message'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:257:in `map_message_to_irc'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/channel.rb:247:in `stream_campfire_to_channel'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/connection.rb:90:in `call'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/connection.rb:90:in `stream'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/connection.rb:103:in `call'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/connection.rb:103:in `<<'
    from /var/lib/gems/1.8/gems/firering-1.2.2/lib/firering/connection.rb:103:in `stream'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:275:in `call'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:275:in `on_decoded_body_data'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:268:in `on_body_data'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:582:in `process_chunk_body'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:330:in `dispatch'
    from /var/lib/gems/1.8/gems/em-http-request-0.3.0/lib/em-http/client.rb:256:in `receive_data'
    from /var/lib/gems/1.8/gems/eventmachine-0.12.10/lib/eventmachine.rb:256:in `run_machine'
    from /var/lib/gems/1.8/gems/eventmachine-0.12.10/lib/eventmachine.rb:256:in `run'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/lib/camper_van/server.rb:21:in `run'
    from /var/lib/gems/1.8/gems/camper_van-0.0.3/bin/camper_van:64
    from /usr/local/bin/camper_van:19:in `load'
    from /usr/local/bin/camper_van:19
```

**Comment by tru** [comment]
Fixed when I updated to ruby1.9
